Formula/vineyard.rb

```diff
@@ -0,0 +1,101 @@
+class Vineyard < Formula
+  include Language::Python::Virtualenv
+
+  desc "In-memory immutable data manager. (Project under CNCF)"
+  homepage "https://v6d.io"
+  url "https://github.com/v6d-io/v6d/releases/download/v0.5.4/v6d-0.5.4.tar.gz"
+  sha256 "ac1d68a7e61778cca737b139fc60a5e05de4f3707b61937ca31a7f9b2c31d21b"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" => :build
+
+  depends_on "apache-arrow"
+  depends_on "boost"
+  depends_on "etcd"
+  depends_on "etcd-cpp-apiv3"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "libgrape-lite"
+  depends_on "open-mpi"
+  depends_on "openssl@1.1"
+  depends_on "tbb"
+
+  fails_with gcc: "5"
+
+  resource "libclang" do
```

Comment:
Isn't this the same as the libclang in llvm?

Reply:
> 

Here the libclang is the python bindings of libclang library.

Reply:
Hi @SMillerDev,

I would like to know if there's any configuration that I can still use the system clang to build rather than the LLVM clang, even I have `llvm` in my dependency?

Here I cannot depends on LLVM, because llvm requires python@3.10 while apache-arrow requires python@3.9, causing a conflict.

But I do need the `libclang.dylib` inside the LLVM formula.

Thanks in advance!

Reply:
I have noticed that once I requires llvm (even non-default llvm, e.g., `llvm@12`), the compiler would be switched to the llvm clang automatically.

Reply:
> I have noticed that once I requires llvm (even non-default llvm, e.g., `llvm@12`), the compiler would be switched to the llvm clang automatically.

This is not true.

We should be able to use the `libclang` Python bindings in the `llvm` formula after #106925. Can you add a comment indicating this?

Reply:
Really happen to see that the issue was fixed from the LLVM side. I would leave a comments here and once LLVM 15 PR been landed we could get rid of the hack there.

Reply:
Not quite fixed upstream, but a workaround that we add here:

https://github.com/Homebrew/homebrew-core/blob/af171627c3b9a381d8aec60d0d458c442650c779/Formula/llvm.rb#L58-L61

Reply:
Thanks for working on it on LLVM itself!